Tbrands 49 single column block refactor

JackHowa opened the pull request:
## Description

Set the stack with the role main as global 16 (desktop) and 11 (default)

- [x] Update block 
- [x] Create token file update in the block storybook
- [x] Create token file update per our process in the source of truth feature pack https://github.com/WPMedia/arc-themes-feature-pack/pull/232

## Jira Ticket

- [TBRANDS-49](https://arcpublishing.atlassian.net/browse/TBRANDS-49)

## Acceptance Criteria

Based on the figma designs apply the global token - spacing-16 as the gap of the single-column-regular to ensure all blocks placed on the page or spaced as per the designs

Set the gap within the Theme file to allow customisation

## Test Steps

### Storybook 

1. Checkout this branch `git checkout TBRANDS-49-single-column-block-refactor`
2. `npm run storybook`
3. Go to the single column layout stories http://localhost:60001/?path=/story/layouts-single-column-regular-layout--layout-with-two-children
4. Notice the gap between the two children match the mock screenshotted in the ticket in desktop breakpoint

![Screen Shot 2022-05-10 at 10 10 30](https://user-images.githubusercontent.com/5950956/167661889-4cb429de-af61-4a88-b882-83b317d4f895.png)

5. Decrease screen size to mobile. Then see the gap size change to 72 px per the mocks 

![Screen Shot 2022-05-09 at 09 53 45](https://user-images.githubusercontent.com/5950956/167437248-6f41caac-1236-4a2c-9f9b-b0929326570f.png)

### Live fusion 

1. `git checkout TBRANDS-49-single-column-layout` feature pack 
2. `npx fusion start -f -l @wpmedia/single-column-layout-block`
3. Create a page with a single column layout
4. Put two blocks within the main to see the gap on mobile and desktop breakpoints
![Screen Shot 2022-05-09 at 10 12 42](https://user-images.githubusercontent.com/5950956/167441038-69530efc-e790-4843-8b11-f0f2add5d9b3.png)
![Screen Shot 2022-05-09 at 10 13 52](https://user-images.githubusercontent.com/5950956/167441153-70dcf685-fca9-4b26-8ab9-0300f7a43bb9.png)

## Effect Of Changes

### Before

Feature blocks within dictate their spacing between each other

### After

Layout blocks dictate the blocks within themselves

## Dependencies or Side Effects

- none

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

JackHowa committed:
feat: Set gap for stack of of single column regular token

Sets the gap in the block token's component styles

JackHowa commented:
Question: 

- In the ticket, I noticed a gap specified for only the regular column's stack. Does this ticket require setting a stack and its gap for the footer and the header as well?

JackHowa committed:
feat: Set 84 px for closest

nschubach commented on blocks/single-column-layout-block/_index.scss:
```diff
@@ -9,6 +9,7 @@
+	@include scss.block-components("single-column-regular");
 }
 
 .b-single-column-regular-footer {
```
Should we BEM these classes?

JackHowa replied:
will work on this, good point 👀

JackHowa replied:
cool updated -- thanks https://github.com/WPMedia/arc-themes-blocks/pull/1334/commits/68e8ea61f0bbfe63e3cd0edfa06f2265bf0a51b4

JackHowa committed:
feat: Add stack along with aria roles for footer and header

JackHowa commented:
> Question:
> 
> * In the ticket, I noticed a gap specified for only the regular column's stack. Does this ticket require setting a stack and its gap for the footer and the header as well?

Update after talking with Nick: 

- We want to make the header and footer to be stacks as well as the parent fragment. 
- I will work on naming these using BEM standards, including the parent `div`

JackHowa committed:
fix: Update bem naming and allow setting the gap on the other containers as well

Per feedback https://github.com/WPMedia/arc-themes-blocks/pull/1334#pullrequestreview-965110575

JackHowa committed:
style: Fix stylelint issues of order include and spacing

JackHowa committed:
feat: Update spacing based on using v1 mocks rather than v3 mocks

Per https://arcpublishing.atlassian.net/browse/TBRANDS-49?focusedCommentId=747140

JackHowa committed:
feat: Update stack to have the same role header as before. Update test description

Previously the test were written for the navigation and the header

JackHowa committed:
docs: Remove interior child component styles and comments

JackHowa committed:
docs: Update peer deps to include arc themes components

JackHowa committed:
perf: Don't publish storybook file

nschubach commented on blocks/single-column-layout-block/_index.scss:
```diff
@@ -1,16 +1,23 @@
+	}
+
+	&__main {
+		max-width: 90rem; // 1440px
+		width: 100%;
```
Should we move this into the theme CSS and use the new alias values or does it make sense to keep this here?

JackHowa replied:
I wasn't sure what was reasonable here for keeping. I can move this to the theme file if that's the recommendation

JackHowa replied:
the alias token layout-max-width?

JackHowa replied:
updated to move the scss to the token file

matthewroach replied:
I don't think we should move this to the token file - we can make an assumption that the width is always 100% - a client could set a max-width to control the maximum width

JackHowa replied:
will revert my move to the theme token file

JackHowa replied:
updated [3f25019](https://github.com/WPMedia/arc-themes-blocks/pull/1334/commits/3f250196c46bc05201a2c880fdf0b6643cb40770) [30ffa8d](https://github.com/WPMedia/arc-themes-blocks/pull/1334/commits/30ffa8d4205f83df37601cac7c260b1502f73afe)

JackHowa committed:
fix: Move all scss into token file and use token

via https://github.com/WPMedia/arc-themes-blocks/pull/1334/files#r869326814

nschubach commented on blocks/single-column-layout-block/layouts/single-column-regular/default.jsx:
```diff
@@ -1,21 +1,33 @@
-			{navigation ? <header className={`${LAYOUT_CLASS_NAME}-header`}>{navigation}</header> : null}
+		<Stack className={LAYOUT_CLASS_NAME}>
+			{navigation ? (
+				<Stack role="banner" className={`${LAYOUT_CLASS_NAME}__header`}>
+					{navigation}
```
We just talked about this in the other ticket, but we should probably make these match as well.

JackHowa replied:
updated https://github.com/WPMedia/arc-themes-blocks/pull/1334/commits/2fbb40d57f73f891ae4946b61e189a87ac530b14

JackHowa committed:
fix: Use pixels for max width of 1440

JackHowa committed:
style: Update naming of the classes to match pagebuilder name

Per https://github.com/WPMedia/arc-themes-blocks/pull/1334#discussion_r869347570

JackHowa committed:
revert: Remove the styles from the token file. Keep in scss block

JackHowa committed:
refactor: Add back styles to the main styles

nschubach commented on blocks/single-column-layout-block/package.json:
```diff
@@ -20,6 +19,7 @@
+    "@wpmedia/arc-themes-components": "*"
   }
-}
+}
```
```suggestion
}

```

nschubach commented on blocks/single-column-layout-block/layouts/.npmignore:
```diff
@@ -1,3 +1,4 @@
-mock*.*
\ No newline at end of file
+mock*.*
+*.story.jsx
```
```suggestion
*.story.jsx

```

Commit:
Update blocks/single-column-layout-block/layouts/.npmignore

Co-authored-by: nschubach <nschubach@gmail.com>

Commit:
Update blocks/single-column-layout-block/package.json

Co-authored-by: nschubach <nschubach@gmail.com>

JackHowa commented:
@nschubach updated extra spacing

nschubach reviewed (approved):
Everything looks good to me.

JackHowa commented:
looks slightly weird in storybook but the same behavior of the 1400 seems to be there
